src/Http/Routing/src/Constraints/FileNameRouteConstraint.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    /// <summary>
+    /// Constrains a route parameter to represent only filename values.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This constraint can be used with a low priority route to disambiguate for requests for
```

Comment:
The level of priority in the route doesn't really matter, does it? The constraint only deals with what the URL looks like, not whether it's useful to prioritize it above or below other routes.

Reply:
I described it here to try and capture the main use case in the docs. I think it's valuable to capture how we expect something to used in a meaningful way. Let me know if you have ideas to improve this.